src/NuGet.Core/NuGet.Packaging/PackageExtractor.cs

```diff
@@ -1088,7 +1088,7 @@ private static void DeleteTargetAndTempPaths(string targetPath, string targetTem
                        token,
                        parentId);
 
-                if (verifyResult.Signed)
+                if (verifyResult.Signed || !verifyResult.Valid)
```

Comment:
I'm concerned that `Valid`, which ideally would be renamed to `IsValid`, doesn't strictly mean that a package is unsigned and yet you're interpreting it to mean just that.  [`PackageSignatureVerifier.VerifySignaturesAsync(…)`](https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.Packaging/Signing/Verification/PackageSignatureVerifier.cs#L35) currently returns `false` for the validity check if a package is unsigned or if the signature is illegal.  This could change in the future.

Reply:
The original idea behind having this if statement was to avoid filling the restore log with unnecessary signature logs for unsigned packages. Therefore even if in the future the definition of validity broadens, it seems correct to have this logs (or error out/throw an exception) if the package is invalid.